api/apiclient.go

```diff
@@ -366,20 +367,19 @@ func (st *state) connectStream(path string, attrs url.Values, extraHeaders http.
 		Proxy:           proxy.DefaultConfig.GetProxy,
 		TLSClientConfig: st.tlsConfig,
 	}
-	var requestHeader http.Header
-	if st.tag != "" {
-		requestHeader = jujuhttp.BasicAuthHeader(st.tag, st.password)
-	} else {
-		requestHeader = make(http.Header)
+	token, err := st.LoginToken()
+	if err != nil {
```

Comment:
The LoginToken method can return the `ErrorTokenNotAvailable` error. Is there a strategy for dealing with that by the caller given we just exit here on any error. What's the expected behaviour.

Reply:
Currently the code that I've seen in Juju always creates a websocket connection to the model first before starting a stream to fetch logs. If you were to create an API connection and set `login: False` then start the stream you'd run into the `ErrorTokenNotAvailable` (which I've now replaced with `ErrLoginFirst`) so I think it's best left to the individual creating the connection to ensure the proper ordering is followed if they use the sessionTokenLoginProvider, I don't think we want to automatically perform a login request in order to obtain the session token.